src/services/sync/strategies/WebUntis/WebUntisSchoolyearSyncer.js

```diff
@@ -0,0 +1,503 @@
+     * * Migrate the changes
+     * * Generate events using text templates
+     * * Send out events for all affected student/teacher/...
+     *
+     * Assumptions:
+     * * Each school has one WebUntis system at most
+     * * This syncer has to be triggered for each school individually
+	 */
+	steps() {
+		this.logInfo(`Running WebUntis Sync in dry run: ${this.dryrun}.\n`);
+
+		/* Why not
+        return this.getWebUntisSystems().then(
+            systems => {
+                // ...
+            }
+        );
+        ? */
+		return this.getUser()
+			.then(user => this.getWebUntisSystems(user))
+			.then(([systems, school]) => {
+				/* TODO: Remove later */
+				// if (systems.length === 0) {
+				// Configure test environment until configuration is available via GUI
+				systems[0] = {
+					alias: 'Test WebUntis',
+					type: 'webuntis',
+					webuntisConfig: {
+						active: true,
+						url: '[URL]',
+						schoolname: '[schoolname]',
+						user: '[username]',
+						password: '[password]',
+					},
+				};
+				// }
+
+				if (systems.length === 0) {
+					return Promise.reject(
+						new Error('No WebUntis configuration for associated school.'),
+					);
+				}
+				this.logInfo(`Found ${systems.length} WebUntis configurations for school ${school.name}.\n`);
+				
+				return Promise.all(systems.map((system) => {
+					this.stats.systems[school.name] = {};
+					return this.syncFromSystem(system, this.stats.systems[school.name], school);
+				}));
+			});
+	}
+
+	async syncFromSystem(system, stats, school) {
+		return this.login(system.webuntisConfig.url, system.webuntisConfig.schoolname,
+			system.webuntisConfig.user, system.webuntisConfig.password)
+			.then(() => this.fetchInformation())
+			.then((data) => {
+				this.logout();
+				stats.temp = data;
+				return Promise.resolve(data);
+			})
+			/*.then(data => {
+				if (this.dryrun) {
+					return this.collectData(data, stats, school);
+				} else {
+					return this.migrateData(data, stats, school);
+				}
+			})*/
+			.then(() => {
+				stats.success = true;
+			});
+	}
+
+	async fetchInformation() {
+		const intermediateData = {};
+		const data = {};
+		data.currentSchoolYear = await this.getCurrentSchoolyear();
+		// intermediateData.holidays = await this.getHolidays();
+		// data.subjects = await this.getSubjects(); // Ignore subjects for now
+		intermediateData.classes = await this.getClasses(data.currentSchoolYear.id);
+		intermediateData.timeGrid = await this.getTimegrid();
+		intermediateData.teachers = await this.getTeachers();
+
+		// data.holidayRanges = data.holidays.map(holiday => [ holiday.startDate, holiday.endDate ]);
+		data.classes = intermediateData.classes.map(klass => ({
+			id: klass.id,
+			name: klass.longName,
+			timetable: [],
+		}));
+		data.timeGrid = intermediateData.timeGrid.map(day => ({
+			day: this.dayLookUp(day.day),
+			timeUnits: day.timeUnits,
+		}));
+
+		if (intermediateData.teachers !== undefined) { // Iterate over teachers
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Block must not be padded by blank lines.](https://app.codacy.com/manual/schul-cloud/schulcloud-server/pullRequest?prid=4061937)